### DISCUSSION TITLE
CableGripper example - error with getObject in step4.py3scn

### DISCUSSION THREAD

**Discussion by cdelor02** [SofaDefrost/SoftRobots#102] [Help - Programming with SoftRobots] [answered]
Hello all. I am trying to use the `step4.py3scn` example from the SoftRobots plugin. I am using SOFA v21.12 on Windows 10, Python version 3.7. However when I run it, I get a number of deprecation notices for the `getObject()` function from `stlib3`, as well as actual errors (see image below) from the SOFA simulation. I imagine the deprecation alerts are a if not the cause of the errors I get, however I don't see in the documentation for STLIB where they mention how `getObject()` is meant to be used, or any indication as to why these soft robot examples shouldn't work with the correct version of Python, STLIB, and SOFA. Thank you.

![step4_py3scn_issues_terminal_only](https://user-images.githubusercontent.com/36216576/154013684-e68fac00-7182-4a16-9ffb-52388b320ac1.png)

**Comment by EulalieCoevoet** [comment, marked as answer]
Hi, 

The problem with `getObject()` has been resolved recently, you can checkout the last version/[release](https://github.com/SofaDefrost/STLIB/releases/tag/release-master) of STLIB.
I've opened a [PR](https://github.com/SofaDefrost/STLIB/pull/66) for the `TetrahedronSetTopologyContainer` errors.
However, those errors should not prevent the simulation from working, so my guess is that it should work for you after updating STLIB.

Hope this helps.
Eulalie

**Reply by cdelor02** [reply]
Thank you. To update STLIB, will I need to redownload the files from the link you provided and rebuild and generate STLIB as I did before? Or can I simply replace the files I have in my SOFA plugin directory? I believe I have the v21.12 from December 16, 2021 (the Windows zip file downloaded from [https://github.com/SofaDefrost/STLIB/releases](https://github.com/SofaDefrost/STLIB/releases)), however you are saying I should download the version named `STLIB_pr-python3variousupdate_for-SOFA-v21.12_Windows.zip` ?

EDIT: Solved by downloading the latest version of STLIB (named above) and replacing old files. Thank you very much for your help @EulalieCoevoet !
